src/Query/QueryBuilder.php

```diff
@@ -302,14 +309,47 @@ public function fetchFirstColumn(): array
      */
     public function executeQuery(): Result
     {
+        [$params, $types] = $this->buildParametersAndTypes();
+
         return $this->connection->executeQuery(
             $this->getSQL(),
-            $this->params,
-            $this->types,
+            $params,
+            $types,
             $this->resultCacheProfile,
         );
     }
 
+    /**
+     * Build then return parameters and types for the query.
+     *
+     * @return array{
+     *     list<mixed>|array<string, mixed>,
+     *     WrapperParameterTypeArray,
+     * } The parameters and types for the query.
+     */
+    private function buildParametersAndTypes(): array
+    {
+        $cteParams = $cteParamTypes = [];
+
+        foreach ($this->commonTableExpressions as $cte) {
+            if (! $cte->query instanceof self) {
+                continue;
```

Comment:
Code coverage complains about this check not beeing covered.

I'd say additional tests using `plain sql strings` as parts instead of QueryBuilder instances should be added anyway, and would cover this line here two (guessing).

Reply:
I've added a test with a sql string as cte query